examples/example-01-basics/src/bin/bp_detect.rs

```diff
@@ -1,16 +1,17 @@
 use libcnb;
-use libcnb::detect::{DetectResult, DetectContext};
 use libcnb::data::build_plan::BuildPlan;
+use libcnb::detect::DetectOutcome;
+use libcnb::detect::GenericDetectContext;
+use libcnb::platform::Platform;
 use libcnb::shared;
-use libcnb::shared::GenericPlatform;
-use libcnb::shared::Platform;
+use std::error::Error;
 
 fn main() {
     libcnb::detect::cnb_runtime_detect(detect)
 }
 
-fn detect(_context: DetectContext<GenericPlatform>) -> DetectResult {
+fn detect(_context: GenericDetectContext) -> Result<DetectOutcome, std::io::Error> {
```

Comment:
Should we return our own Error type here?

Reply:
It is a custom error type defined here: https://github.com/Malax/libcnb/pull/2/files#diff-c11e72174294250d95d68429cbb5e9ec58c16ac64087d428736be92bd22cd688R24-R26

I wanted it to be flexible so buildpack authors don't need to know anything about our error types. We can also implement the `BuildpackError` for other Error libraries if we need to. Using `io::Error` here was basically just a test if that works as I expected.

Do you want me to change the example?

Reply:
nah, I don't care about custom error types on the example. I expect a buildpack author to be "sloppy" and do convenience things to start.